crates/pgt_workspace/src/code_actions.rs

```diff
@@ -0,0 +1,65 @@
+use crate::workspace::StatementId;
+use pgt_configuration::RuleSelector;
+use pgt_fs::PgTPath;
+use pgt_text_size::TextSize;
+
+#[derive(Debug, serde::Serialize, serde::Deserialize)]
+#[cfg_attr(feature = "schema", derive(schemars::JsonSchema))]
+pub struct CodeActionsParams {
+    pub path: PgTPath,
+    pub cursor_position: TextSize,
+    pub only: Vec<RuleSelector>,
+    pub skip: Vec<RuleSelector>,
+}
+
+#[derive(Debug, serde::Serialize, serde::Deserialize)]
+#[cfg_attr(feature = "schema", derive(schemars::JsonSchema))]
+pub struct CodeActionsResult {
+    pub actions: Vec<CodeAction>,
+}
+
+#[derive(Debug, serde::Serialize, serde::Deserialize)]
+#[cfg_attr(feature = "schema", derive(schemars::JsonSchema))]
+pub struct CodeAction {
+    pub title: String,
+    pub kind: CodeActionKind,
+    pub disabled_reason: Option<String>,
+}
+
```

Comment:
maybe we can put this in to a `features/` directory? we could also structure the other feature params and types in a similar manner.